src/pthreads.c

```diff
@@ -54,6 +85,7 @@
 #include "threadpool-object.h"
 #include "threadpool-utils.h"
 
+thread_local size_t capped_num_threads = UINT_MAX;
```

Comment:
Thread-local cap doesn't make sense, the same thread can work with different `pthreadpool_t` objects and multiple threads can submit tasks to the same `pthreadpool_t` object.

Limit on the number of threads should be a property of `pthreadpool_t` object, not of a thread.

Reply:
I think that seems fair. I had something else in mind, but what you said makes more sense.

Reply:
Actually for the case of "multiple threads can submit tasks to the same pthreadpool_t object ", the issue is this. If `capped_num_threads` was part of `pthreadpool_t` object then we would have to atomically update it and depending on whose update finishes last we might apply different caps. So that may break when thread 1 sets cap to 2 but later on it gets set to 3 by thread 2 and then thread 1 runs with the cap of 3 threads and so does thread 2. In order to make this work we will have to change `*parallelize*` API to account for cap.

But for "the same thread can work with different pthreadpool_t objects", what you said makes sense.

Dont have a good solution but did want to point it out in case I missed something.